doc/api/http.md

```diff
@@ -1277,6 +1277,13 @@ const server = http.createServer((req, res) => {
 });
 ```
 
+If [`response.writeHead()`][] method is called and this method has not been
+called, it will directly write the supplied header values onto the network
+channel without caching internally, and the [`response.getHeader()`][] on the
+header will not yield the expected result. If progressive population of headers
+is desired with potential future retrieval and modification, use
```

Comment:
Is modification still possible after `writeHead`? Sounds to me like only retrieval is possible. Ditto below.

Reply:
@TimothyGu - yes, that is right: headers written through `writeHead` are not modifiable, and attempt to do so results in  error `Cannot set headers after they are sent to the client`. Hence the suggestion to use `setHeader()` API for such cases. If the text is not conveying that let me know where and how can I fix it, thanks.

Reply:
Yeah, you wrote

> If progressive population of headers is desired with potential future retrieval **and modification**, use …

It seemed to me that the bolded bit should perhaps be removed.

Reply:
sorry, missed this update.  Sorry, I am missing your point:
#cat 10354.js 
```js
const h = require('http')
h.createServer((q, r) => {
  r.setHeader('foo', 'bar')
  r.setHeader('foo', 'new')
  r.end()
}).listen(12000, () => {
  const c = h.get('http://localhost:12000', (m) => {
    console.log(m.headers)
  })
})
```
#node 10354.js
```
{ foo: 'new',
  date: 'Thu, 21 Jun 2018 14:42:33 GMT',
  connection: 'close',
  'content-length': '0' }
```

`setHeader()` is indeed used for modification, and hence the text. Can you please clarify your point?

Reply:
Never mind, I see what this is trying to say now.